docs/best_practices.rst

```diff
@@ -0,0 +1,57 @@
+#######################################
+edx-platform Development Best Practices
+#######################################
+
+There are many general best practices documented for `Open edX Development in
+Confluence`_. The following best-practices are specific to edx-platform.
+
+Course Access in LMS
+********************
+
+the following technologies can be used to access course-related data:
+
+* `Course Overviews`_:  Provide performant access to course metadata.
+
+* `Course Blocks`_: Provide performant access to the blocks in a course,
+  including filtering and access control capabilities.
+
+* `Modulestore`_ - Contains all course related data, including course metadata
+  course blocks, and student module data. Course Overviews and Course Blocks are
+  cached performant versions of a subset of this source data.
+
+When coding in the LMS, it is generally preferred to use `Course Overviews`_ and
+`Course Blocks`_, due to the following benefits:
+
+1. Cached versions of course data that are better optimized for Learners.
+
+2. A start of the separation of LMS and Studio data to move us closer to the
+   ultimate ability to separate the two.
+
+**Note**: At this time, these preferred methods are for coding in the LMS, but
+   outside of the courseware.  Inside the courseware, there is more work to be
+   done to take advantage of these techniques.
+
+Prefer using `Course Overviews`_ where possible when you just need course
+metadata, rather than loading the full course. For example, this could be done
+by calling ``get_course_overview_with_access()`` in place of
```

Comment:
Would it make sense to list out some examples of metadata that Course Overviews provide?